src/features/auth/guards/auth.guard.ts

```diff
@@ -16,7 +16,15 @@ export class JwtAuthGuard extends AuthGuard('jwt') {
       context.getClass(),
     ]);
 
-    if (isPublic) {
+    const allowUnauthenticatedRequest =
+      this.reflector.getAllAndOverride<boolean>(
+        IS_ALLOW_UNAUTHENTICATE_REQUEST,
+        [context.getHandler(), context.getClass()],
+      );
+
+    const req = context.switchToHttp().getRequest();
+    const isAllowed = allowUnauthenticatedRequest && !req.headers.authorization;
```

Comment:
What do you think about this?

```suggestion
    const isUnauthenticatedRequest = allowUnauthenticatedRequest && !req.headers.authorization;
```